index.js

```diff
@@ -0,0 +1,219 @@
+// GET AFRICA
+const africaTrips = () => {
+  let url = (URL + '/continent?query=Africa')
+  getTrips(url)
+}
+// GET Antartica
+const antarticaTrips = () => {
+  let url = (URL + '/continent?query=Antartica')
+  getTrips(url)
+}
+// GET Australasia
+const australasiaTrips = () => {
+  let url = (URL + '/continent?query=Australasia')
+  getTrips(url)
+}
+// GET EUROPE
+const europeTrips = () => {
+  let url = (URL + '/continent?query=Europe')
+  getTrips(url)
+}
+// GET North America
+const nAmericaTrips = () => {
+  let url = (URL + '/continent?query=North%20America')
+  getTrips(url)
+}
+// GET South America
+const sAmericaTrips = () => {
+  let url = (URL + '/continent?query=South%20America')
+  getTrips(url)
+}
+
+// GET details for single trip
+const getTripDetails = function getTripDetails(id) {
+
+  let tripDetail = $('#trip-detail-section');
+  tripDetail.empty();
+
+  axios.get(URL + id)
+    .then((response) => {
+      let data = response.data;
+      let name = $(`<h4><strong>Name:</strong> ${data.name}</h4>`).addClass(`${id}`);
+      let about = $(`<span><strong>Description:</strong> ${data.about.slice(0, 150)}</span>`).addClass("teaser");
+      let aboutFullText = $(`<span>${data.about.slice(150)}</span>`).addClass("moreinfo hidden").attr('id', 'info1');
+      let more = $(`<span>...Read more</span>`).addClass("more").attr('target', 1);
+      let continent = $(`<p><strong>Continent:</strong> ${data.continent}</p>`);
+      let category = $(`<p><strong>Category:</strong> ${data.category}</p>`);
+      let weeks = $(`<p><strong>Weeks:</strong> ${data.weeks}</p>`);
+      let cost = $(`<p><strong>Cost:</strong> $${data.cost}</p>`);
+
+      tripDetail.append(name, about, aboutFullText, more, continent, category, weeks, cost);
+
+      reportStatus(`Successfully loaded details for ${response.data.name} trip`)
+    })
+    .catch((error) => {
+      console.log(error);
+      reportStatus(`Error: ${error.message}`);
+    });
+}
+
+// Form helpers
+const FORM_FIELDS = ['name', 'email'];
+const inputField = name => $(`#trip-form input[name="${name}"]`);
+
+const readFormData = () => {
+  const getInput = name => {
+    const input = inputField(name).val();
+    return input ? input : undefined;
+  };
+
+  const formData = {};
+  FORM_FIELDS.forEach((field) => {
+    formData[field] = getInput(field);
+  });
+
+  return formData;
+};
+
+const clearForm = () => {
+  FORM_FIELDS.forEach((field) => {
+    inputField(field).val('');
+  });
+}
+
+// Reserve spot on trip
+const reserveTrip = (event) => {
+  event.preventDefault();
+  // let id = Number($('#trip-detail-section h4')[0].classList[0]);
+  let id = $('.trip-detail-section h4').attr("class");
+
+  const tripData = readFormData();
+
+  reportStatus('Sending trip reservation data...');
+
+  axios.post((`${URL}${id}/reservations`), tripData)
+    .then((response) => {
+      clearForm();
+      reportStatus(`Successfully created trip reservation with ID ${response.data.trip_id}!`);
```

Comment:
`response.data.trip_id` refer to the ID of the trip, not the reservation itself. Funnily enough, the API doesn't give back a response with the reservation ID!